azure/Guidelines.md

```diff
@@ -846,109 +848,189 @@ https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#perfo
+```
 
-<a href="#lro-put-valid-inputs-synchronously" name="lro-put-valid-inputs-synchronously">:white_check_mark:</a> **DO** perform as much validation as practical when initiating the operation to alert clients of errors early.
+- The request body identifies parameters required to execute the action.
 
-<a href="#lro-put-returns-200-or-201" name="lro-put-returns-200-or-201">:white_check_mark:</a> **DO** return a `201-Created` status code for create or `200-OK` for replace from the initial request with a representation of the resource if the resource was created successfully.
+- This operation implicitly creates a status monitor resource and sets its `kind` property to something like `<ResourceType><action>`; for example `VMRebooting`.
 
-<a href="#lro-put-returns-operation-id-header" name="lro-put-returns-operation-id-header">:white_check_mark:</a> **DO** include an `Operation-Id` header in the response with the ID of the status monitor for the operation.
+- The response body must be the implicily created status monitor resource.
 
-<a href="#lro-put-response-headers" name="lro-put-response-headers">:white_check_mark:</a> **DO** include response headers with any additional values needed for a GET request to the status monitor (e.g. location).
+- The `operation-id` header allows the client to specify the status monitor's resource ID and is also used for retries/idempotency. If unspecified, the service may create an operation-id (typically a GUID) and return it via the `operation-id` and `operation-location` response headers; in this case, the service must figure out how to deal with retries/idempotency.
 
-<a href="#lro-put-returns-operation-location" name="lro-put-returns-operation-location">:ballot_box_with_check:</a> **YOU SHOULD** include an `Operation-Location` header in the response with the absolute URL of the status monitor for the operation.
+  - For an idempotent POST (same `operation-id` or same request body within some short time window), the service should return the same response as shown above.
 
-<a href="#lro-put-operation-location-includes-api-version" name="lro-put-operation-location-includes-api-version">:ballot_box_with_check:</a> **YOU SHOULD** include the `api-version` query parameter in the `Operation-Location` header with the same version passed on the initial request if it is required by the get operation on the status monitor.
+  - For a non-idempotent POST, the service can treat the POST operation as idempotent (if performed within a short time window) or can treat the POST operation as initiating a brand new LRO action operation.
 
-#### Obtaining status and results of long-running operations
+<a href="#lro-action-no-resource" name="lro-action-no-resource">:white_check_mark:</a> **DO** use the following pattern when implementing an LRO action not related to a specific resource (such as a batch operation):<p>
+```
+PUT /operations/<operation-id>
+<JSON Status Monitor Resource in body; `kind` MUST be set>
+```
```

Comment:
This seems to be the VLRO we discussed (on whether it takes long time or not). But here we are framing it on whether it is about a resource or not?

Reply:
Yes, we've shifted focus a bit. These guidelines should work for many of the VLROs we have wrestled with recently, and they are easier to explain.